pkg/certificate/mrc_reconciler.go

```diff
@@ -1,214 +1,182 @@
-		return false
-	}
+	return m.updateIssuers(issuer, issuer)
+}
 
-	//The truncate is needed so we truncate the monotonic clock.
-	return transitionAfter.Truncate(0).After(time.Now())
+var validMRCIntentCombinations = map[v1alpha2.MeshRootCertificateIntent][]v1alpha2.MeshRootCertificateIntent{
+	v1alpha2.ActiveIntent: {
+		v1alpha2.PassiveIntent,
+		v1alpha2.ActiveIntent,
+	},
+	v1alpha2.PassiveIntent: {
+		v1alpha2.ActiveIntent,
+	},
 }
 
-// shouldUpdateState (and conditions).
-func shouldUpdateState(mrc *v1alpha2.MeshRootCertificate) bool {
-	// TODO(5046): determine the status to we need to be at to be able to move forward.
-	var status v1alpha2.MeshRootCertificateComponentStatus
-
-	for _, useCase := range allUseCases {
-		switch useCase {
-		case UseCaseValidatingWebhook:
-			if mrc.Status.ComponentStatuses.ValidatingWebhook != status {
-				return false
-			}
-		case UseCaseMutatingWebhook:
-			if mrc.Status.ComponentStatuses.MutatingWebhook != status {
-				return false
-			}
-		case UseCaseXDSControlPlane:
-			if mrc.Status.ComponentStatuses.XDSControlPlane != status {
-				return false
-			}
-		case UseCaseSidecar:
-			if mrc.Status.ComponentStatuses.Sidecar != status {
-				return false
-			}
-		case UseCaseBootstrap:
-			if mrc.Status.ComponentStatuses.Bootstrap != status {
-				return false
-			}
-		case UseCaseGateway:
-			if mrc.Status.ComponentStatuses.Gateway != status {
-				return false
-			}
-		}
+// ValidateMRCIntents validates the intent combination of MRCs
+func ValidateMRCIntents(mrc1, mrc2 *v1alpha2.MeshRootCertificate) error {
+	if mrc1 == nil || mrc2 == nil {
+		msg := "unexpected nil MRC provided when validating MRC intents"
+		log.Error().Msg(msg)
+		return fmt.Errorf(msg)
 	}
-	return true
-}
 
-// update the MRCComponentStatus
-func (m *Manager) updateMRCComponentStatus(mrc *v1alpha2.MeshRootCertificate) error {
-	// TODO(5046): determine the status to set the owned components to.
-	var status v1alpha2.MeshRootCertificateComponentStatus
-
-	for _, useCase := range m.ownedUseCases {
-		switch useCase {
-		case UseCaseValidatingWebhook:
-			// validating, issuing unknown.
-			mrc.Status.ComponentStatuses.ValidatingWebhook = status
-		case UseCaseMutatingWebhook:
-			mrc.Status.ComponentStatuses.MutatingWebhook = status
-		case UseCaseXDSControlPlane:
-			mrc.Status.ComponentStatuses.XDSControlPlane = status
-		case UseCaseSidecar:
-			mrc.Status.ComponentStatuses.Sidecar = status
-		case UseCaseBootstrap:
-			mrc.Status.ComponentStatuses.Bootstrap = status
-		case UseCaseGateway:
-			mrc.Status.ComponentStatuses.Gateway = status
+	intent1 := mrc1.Spec.Intent
+	intent2 := mrc2.Spec.Intent
+
+	validIntents, ok := validMRCIntentCombinations[intent1]
+	if !ok {
+		log.Error().Err(ErrUnknownMRCIntent).Msgf("unable to find %s intent in set of valid intents. Invalid combination of %s intent and %s intent", intent1, intent1, intent2)
+		return ErrUnknownMRCIntent
+	}
+
+	for _, intent := range validIntents {
```

Comment:
should this be a map to check for access?

Reply:
Are you suggesting using a nested map for `validMRCIntentCombinations`?

Reply:
yup

Reply:
Updated!